adapters/nextmillennium/nextmillennium.go

```diff
@@ -128,13 +136,22 @@ func createBidRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext
 	ext := nextMillJsonExt{}
 	ext.Prebid.StoredRequest.ID = placementID
 	ext.NextMillennium.NmmFlags = flags
-	jsonExt, err := json.Marshal(ext)
+	bidRequest := *prebidBidRequest
+	jsonExtCommon, err := json.Marshal(ext)
 	if err != nil {
 		return prebidBidRequest
 	}
-	bidRequest := *prebidBidRequest
+	bidRequest.Imp[0].Ext = jsonExtCommon
+	ext.Prebid.Server = &serverOverloader{
+		GvlID:       serverParams.GvlID,
+		DataCenter:  serverParams.DataCenter,
+		ExternalUrl: serverParams.ExternalUrl,
+	}
+	jsonExt, err := json.Marshal(ext)
```

Comment:
Why are you marshalling this structure twice?

Reply:
The ext structure is also used in imp.ext, but we don't need the server structure in imp.ext, only in ext. I wasn't sure which way would be better: marshal twice or delete the field using jsonparser.Delete()